perun-core/src/main/java/cz/metacentrum/perun/core/impl/MembersManagerImpl.java

```diff
@@ -802,4 +811,53 @@ public void setSponsoredAccountsConfigLoader(SponsoredAccountsConfigLoader spons
 	public NamespaceRules getNamespaceRules(String namespace) throws NamespaceRulesNotExistsException {
 		return sponsoredAccountsConfigContainer.getNamespaceRules(namespace);
 	}
+
+	@Override
+	public void rejectAllMemberOpenApplications(PerunSession sess, Member member) {
+		try {
+			List<Integer> ids = jdbc.query(
+				"select id from application " +
+					"where user_id=? and vo_id=? and state not in (?, ?)",
+				new SingleColumnRowMapper<>(Integer.class),  member.getUserId(), member.getVoId(), rejected, approved);
```

Comment:
I was testing this on devel and I have discovered a problem. If a user wants to submit a group application, and he is not in the Perun yet, the group application doesn't contain `user_id`. Therefore, this won't work, because the application will not be found. :disappointed: 

I'm not sure how to exactly solve this. We might set the `user_id` for all user applications, when the object user is created. Or, you would need to search the applications here based on extsources (which, I guess, would be much difficult). What do you think? :thinking: